test/e2e/framework/test_context.go

```diff
@@ -436,4 +437,46 @@ func AfterReadingAllFlags(t *TestContextType) {
 		}
 		os.Exit(1)
 	}
+
+	// Check for num-nodes if not set by looking for all nodes that are schedulable
+	if t.CloudConfig.NumNodes == -1 {
+		numNodes, err := countScheduleableNodes(t.KubeContext, t.KubeConfig, t.Host)
+		if err != nil {
+			klog.Errorf("Error getting num-nodes from API: %v", err)
+		} else {
+			fmt.Println("numNodes: ", numNodes)
+			t.CloudConfig.NumNodes = numNodes
+		}
+	}
+}
+
+// Gets a config used to create a client
+// TODO(stevesloka) We should simplify this across the framework and not duplicate
+func loadConfig(kubeContext, kubeconfig, host string) (*restclient.Config, error) {
```

Comment:
You should not need this in the framework, it should exist by the time this is called.

Reply:
From what I can see, the client is regenerated for each test (https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L187). 

Is there a spot where this exists and I can hook in? The current place in `AfterReadingAllFlags` doesn't have this initialized yet.